kmcd-props.md

```diff
@@ -23,17 +23,19 @@ Measurables
 ```k
     syntax Event ::= Measure
     syntax Measure ::= Measure () [function]
-                     | Measure ( debt: Rat , controlDai: Map , potChi: Rat , potPie: Rat , sumOfScaledArts: Rat, vice: Rat, endDebt: Rat )
- // --------------------------------------------------------------------------------------------------------------------------------------
-    rule [[ Measure() => Measure(... debt: DEBT, controlDai: controlDais(keys_list(VAT_DAIS)), potChi: POT_CHI, potPie: POT_PIE, sumOfScaledArts: calcSumOfScaledArts(VAT_ILKS, VAT_URNS), vice: VAT_VICE, endDebt: END_DEBT) ]]
-         <vat-debt> DEBT     </vat-debt>
-         <vat-dai>  VAT_DAIS </vat-dai>
-         <vat-ilks> VAT_ILKS </vat-ilks>
-         <vat-urns> VAT_URNS </vat-urns>
-         <vat-vice> VAT_VICE </vat-vice>
-         <pot-chi>  POT_CHI  </pot-chi>
-         <pot-pie>  POT_PIE  </pot-pie>
-         <end-debt> END_DEBT </end-debt>
+                     | Measure ( debt: Rat , controlDai: Map , potChi: Rat , potPie: Rat , sumOfScaledArts: Rat , vice: Rat , endDebt: Rat , sumOfAllFlapLots: Rat , dai: Map , sumOfAllFlapBids: Rat , mkrBalances: Map )
+ // ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+    rule [[ Measure() => Measure(... debt: DEBT, controlDai: controlDais(keys_list(VAT_DAIS)), potChi: POT_CHI, potPie: POT_PIE, sumOfScaledArts: calcSumOfScaledArts(VAT_ILKS, VAT_URNS), vice: VAT_VICE, endDebt: END_DEBT, sumOfAllFlapLots: sumOfAllFlapLots(FLAP_BIDS), dai: VAT_DAIS, sumOfAllFlapBids: sumOfAllFlapBids(FLAP_BIDS), mkrBalances: MKR_BALS) ]]
+         <vat-debt>     DEBT      </vat-debt>
+         <vat-dai>      VAT_DAIS  </vat-dai>
+         <vat-ilks>     VAT_ILKS  </vat-ilks>
+         <vat-urns>     VAT_URNS  </vat-urns>
+         <vat-vice>     VAT_VICE  </vat-vice>
+         <pot-chi>      POT_CHI   </pot-chi>
+         <pot-pie>      POT_PIE   </pot-pie>
+         <end-debt>     END_DEBT  </end-debt>
+         <flap-bids>    FLAP_BIDS </flap-bids>
+         <gem-balances> MKR_BALS  </gem-balances>
```

Comment:
One problem with this is that the cell `<gem-balances>` shows up inside a `multiplicity="*" type="Map"` cell.

From `gem.md`:

```
    configuration 
      <gems> 
        <gem multiplicity="*" type="Map"> 
          <gem-id>       "":String </gem-id> 
          <gem-wards>    .Set      </gem-wards> 
          <gem-balances> .Map      </gem-balances> // mapping (address => uint256) Address |-> Wad 
        </gem> 
      </gems> 
```

So this will actually just be picking out some _random_ gem to get the balances cell from, not necessarily the `MKR` gem.

You want to say something more like this:

```
<gem> <gem-id> "MKR" </gem-id> <gem-balances> MKR_BALS </gem-balances> ... </gem>
```

This may fix your issue because when you have a `type="Map"` cell, it expects that _every_ rule which mentions any sub-cells of it _also_ mentions the "key" cell, which is the first sub-cell (in this case `<gem-id>` cell). Internally, this is used as an optimization to turn this access into a map lookup instead of a randomized search through all potential cells.

Reply:
Doing that at least changed the error message:
```
%map* (%block*)* bitcast (%block* (%block*)* @apply_rule_4865 to %map* (%block*)*)/mnt/stuff/maker/mkr-mcd-spec/deps/k/k-distribution/target/release/k/lib/../bin/llvm-kompile: line 27: 26700 Aborted                 (core dumped) "$(dirname "$0")"/llvm-kompile-codegen "$definition" "$dt_dir"/dt.yaml "$dt_dir" $debug > "$mod"
```

Reply:
For the record: was a bug in K, fixed with deps update.